pkg/operatorstatus/operatorstatus.go

```diff
@@ -133,11 +133,18 @@ func (o *ClusterOperatorHandler) setOperatorStatus(condtype configv1.ClusterStat
 			Message:            msg,
 			LastTransitionTime: metaapi.Now(),
 		})
-		if !modified && state.Status.Version == currentVersion {
+		if !modified && len(state.Status.Versions) > 0 && state.Status.Versions[0].Version == currentVersion {
```

Comment:
all this logic is very much a short term hack that assumes we only ever put a single entry in Status.Versions.  If we ever need more, this will need revisiting.